docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -56,48 +56,114 @@ Each component is composed of several different parts.
 These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
 You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
 
+### Overrides based on props
+
+Each slot of the component's `styleOverrides` support a key of `variants`. Variants are sets of overrides when the specified props are matched.
+
+The definitions are specified in an array with sensitive order, so make sure that the styles that should win are specified last.
```

Comment:
Maybe we can omit the "sensitive order" bit, because by itself doesn't mean anything. The next sentence already gives the necessary information.

```suggestion
The definitions are specified in an array. Make sure the styles that should take precedence are listed last.
```